### PR TITLE
feat: `networks run` CLI [APE-981]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

A CLI for starting development nodes

fixes: #1453
fixes: https://github.com/ApeWorX/ape-hardhat/issues/128
Fixes: APE-981
Fixes: APE-725

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape_networks/_cli.py [hunk 1]
```diff
@@ -73,3 +75,31 @@ def make_sub_tree(data: Dict, create_tree: Callable) -> Tree:
...
+@network_option(default="ethereum:local:geth")
+def run(cli_ctx, network):
+    """
+    Start a node process
+    """
+    network_ctx = cli_ctx.network_manager.parse_network_choice(network)
+    provider = network_ctx._provider
+    if not isinstance(provider, SubprocessProvider):
+        cli_ctx.abort(
+            f"`ape-node` requires a provider that manages a process, not '{provider.name}'."
```
Comment (fubuloubu) [inline src/ape_networks/_cli.py:91]:
```suggestion
            f"`ape networks run` requires a provider that manages a process, not '{provider.name}'."
```

File: tests/integration/cli/test_networks.py [hunk 2]
```diff
@@ -129,3 +129,13 @@ def test_filter_networks(ape_cli, runner, networks):
...
     assert_rich_text(result.output, _TEST_PROVIDER_TREE_OUTPUT)
+
+
+@run_once
+def test_node_not_subprocess_provider(ape_cli, runner):
+    result = runner.invoke(ape_cli, ["networks", "run", "ethereum:local:test"])
+    assert result.exit_code != 0
+    assert (
+        result.output
+        == "ERROR: `ape-node` requires a provider that manages a process, not 'test'.\n"
```
Comment (fubuloubu) [inline tests/integration/cli/test_networks.py:140]: Same here

**Review by fubuloubu — DISMISSED** [review]
Looks really good! Two comments

**Comment by antazoey** [comment]
- [ ] needing to do some adjustments as I actually use this, lol

**Comment by antazoey** [comment]
Screenshot 2023-08-01 at 11 08 05

Now your output looks identical as if you were running the process yourself!

**Review by fubuloubu — DISMISSED** [review]

**Pull request merged**